src/main/java/com/ryanharter/autoparcel/AutoParcelExtension.java

```diff
@@ -51,16 +48,20 @@ public boolean mustBeAtEnd(Context context) {
   }
 
   @Override
-  public String generateClass(final Context context, final String className, final String classToExtend, boolean isFinal) {
+  public String generateClass(final Context context, final String className,
+                              final String classToExtend, boolean isFinal) {
     return getParcelableCode(context, className, classToExtend, context.properties());
   }
 
   String getParcelableCode(Context context, String className, String classToExtend,
-                                   Map<String, ExecutableElement> properties) {
+                           Map<String, ExecutableElement> properties) {
+    validateProperties(context.processingEnvironment(), properties);
```

Comment:
You know more about the AutoValue part, but I think I would move this to happen in `applicable` so that validation controls whether or not your return `true` from that method. By calling `printMessage` with an `ERROR` kind you're going to fail the compilation so you won't need to use the exception for control flow and you can skip having to propagate and exception up through AutoValue.